apps/wallet-mock/src/test/dapp-connect.test.ts

```diff
@@ -0,0 +1,417 @@
+    await expect(page.getByTestId('dapp-select-wallet-modal')).toBeVisible()
+    await expect(page.getByTestId('dapp-select-wallet-modal')).toContainText(
+      'Connect to website'
+    )
+  })
+
+  test('should see hostname', async () => {
+    await expect(page.getByTestId('dapp-select-hostname')).toBeVisible()
+    await expect(page.getByTestId('dapp-select-hostname')).toHaveText(
+      connectionContent.hostname
+    )
+  })
+
+  test('should see wallet to connect form', async () => {
+    await expect(page.getByTestId('dapp-select-wallet-form')).toBeVisible()
+    await expect(page.getByTestId('dapp-select-wallet-form')).toContainText(
+      'Select a wallet to connect to'
+    )
+  })
+
+  test('should see wallets list', async () => {
+    await Promise.all(
+      selectionContent.availableWallets.map((wallet) =>
+        expect(
+          page.getByTestId('dapp-select-wallet-form').getByLabel(wallet)
+        ).toBeVisible()
+      )
+    )
+  })
+
+  test('should see approve button', async () => {
+    await expect(page.getByTestId('dapp-select-approve-button')).toBeVisible()
+    await expect(page.getByTestId('dapp-select-approve-button')).toHaveText(
+      'Approve'
+    )
+  })
+
+  test('should see approve button disabled', async () => {
+    await expect(page.getByTestId('dapp-select-approve-button')).toBeDisabled()
+  })
+
+  test('should see deny button', async () => {
+    await expect(page.getByTestId('dapp-select-deny-button')).toBeVisible()
+    await expect(page.getByTestId('dapp-select-deny-button')).toHaveText('Deny')
+  })
+
+  test('percy snapshot', async () => {
+    await percySnapshot(page, 'connect_dapp_wallet_selection')
+  })
+})
+
+test.describe('Connect to Dapp - wallet selection actions', () => {
+  let page: Page
+  test.beforeEach(async ({ browser }) => {
+    page = await browser.newPage()
+    await mock(page)
+    await page.goto('/')
+    await beginInteractionSession(page, 'WALLET_CONNECTION')
+    await sendBackendInteraction(
+      page,
+      'REQUEST_WALLET_CONNECTION_REVIEW',
+      connectionContent
+    )
+    await sendBackendInteraction(
+      page,
+      'REQUEST_WALLET_SELECTION',
+      selectionContent
+    )
+  })
+
+  test.afterEach(async () => {
+    await endInteractionSession(page)
+  })
+
+  test('should close modal on deny', async () => {
+    await page.getByTestId('dapp-select-deny-button').click()
+    await expect(page.getByTestId('dapp-select-wallet')).toHaveCount(0)
+  })
+
+  test('should enable approve button when wallet chosen', async () => {
+    await page
+      .getByTestId('dapp-select-wallet-form')
+      .getByRole('radio')
+      .first()
+      .click()
+    await expect(page.getByTestId('dapp-select-approve-button')).toBeEnabled()
+  })
+
+  test.fixme('should close modal on approve', async () => {
+    //TODO: add approve action logic, add test for approve action
+  })
+})
+
+test.describe('Connect to Dapp - passphrase request validations', () => {
```

Comment:
It may not be possible but can we try and see if we can tests for an incorrect passphrase as well?

Reply:
Not for now, I think your fixes for interaction flow are needed, as for now nothing happens when you click approve. I will lave this test blank to be created, after fix.